### ISSUE TITLE
Create room within signaling server

### ISSUE THREAD

**Issue by felix28** [muaz-khan/RTCMultiConnection#511]
For example, User A opens a rooms. User B, User C and User D joins, everything is fine. However, when User A leaves, he cannot rejoin in the room and other users cannot enter.

I'm thinking that one way to solve this is by creating the room within the signaling server. We need something that the room is open forever and all users in the room can just join and leave as they want.

How to solve this?

Thanks

**Comment by stantonsa** [comment]
+1

**Comment by Wemperer** [comment]
webRTC is peer to peer. It makes sense that if the host leaves, the room goes with him. Hosting the room on the signaling server defeats the purpose of p2p. A better solution would be to have the host and guests switch to someone who is already in the room. You should be able to implement that yourself.

Host Leave -> other users detect host leaves -> select new host -> host spins up room -> others join

**Comment by felix28** [comment]
Thank you. Do you have an idea how I can achieve this? I will try to do this but I will appreciate if you can provide me a source code on how to make this happen.

Thanks

**Comment by JM-Mendez** [comment]
If you want to create a room that does not disconnect when the moderator leaves, you need to specify it like so

```js
const connection = new RTCMultiConnection()
connection.autoCloseEntireSession = false
```
https://github.com/muaz-khan/RTCMultiConnection/blob/ed6a144af688ebd5744a81095787e0b9351a313d/docs/api.md

**Comment by duahimanshu100** [comment]
Any solutions to this yet? Its not working. Moderator can't join the room again. Instead it creates the room again with same name.

**Comment by asfandyardigital** [comment]
`    beforeOpenOrJoin(roomid, function() {
      connection.checkPresence(roomid, function(isRoomExist, roomid) {
        // join callback
        if (isRoomExist === true) {
          connection.join(roomid);
        } else {
          connection.open(roomid);
        }
        afterOpenOrJoin();
      });
    });`
You can use **connection.checkPresence** function to check if room is available or not . If it is available you can join the room else create a new one ..
On upon **leaving room** use this code.
`  // disconnect with all users
    connection.getAllParticipants().forEach(function(pid) {
      connection.disconnectWith(pid);
    });

    // stop all local cameras
    connection.attachStreams.forEach(function(localStream) {
      localStream.stop();
    });

    // close socket.io connection
    connection.closeSocket();`
This will close the socket connection of moderator and he can join the room again.

**Comment by muaz-khan** [comment]
This issue is solved since [3.4.7](https://github.com/muaz-khan/RTCMultiConnection/releases/tag/3.4.7).  Ref: https://github.com/muaz-khan/RTCMultiConnection/commit/7e35a011c305696fe0d932c2c0edd1dae0be04ad (and all upcoming commits)

[NPM package](https://www.npmjs.com/package/rtcmulticonnection) or `git clone` is recommended to fetch/get latest release. Current version is [3.5.2](https://www.npmjs.com/package/rtcmulticonnection?activeTab=versions).

[Signaling-server](https://www.npmjs.com/package/rtcmulticonnection-server) is also isolated to be installed separately, if needed.
